tests/test_connection.py

```diff
@@ -0,0 +1,52 @@
+import pytest
+from trio_websocket import ConnectionClosed, WebSocketClient, WebSocketServer
+import trio
+
+
+import logging
+logging.basicConfig(level=logging.DEBUG)
+HOST = 'localhost'
+RESOURCE = 'resource'
+
+
+@pytest.fixture
+async def echo_server(nursery):
+    async def handler(conn):
+        try:
+            msg = await conn.get_message()
+            await conn.send_message(msg)
+        except ConnectionClosed:
+            pass
+    server = WebSocketServer(handler, HOST, 0, ssl_context=None)
+    await nursery.start(server.listen)
+    yield server
+
+
+def client_for_server(server):
```

Comment:
you can use another, dependent fixture here:
```
async def client(echo_server):
    client = WebSocketClient(...)
    yield client
    client.close()
```
which makes me wonder:
   * does client not have a close() method?
   * is the library providing an async contextmanager for these clients?

Ideally the fixture code would be:
```
async with open_client(...) as client:
    yield client
```